src/test/java/com/microsoft/sqlserver/jdbc/datatypes/SQLServerSpatialDatatypeTest.java

```diff
@@ -51,141 +52,1161 @@
+        assertTrue(Arrays.equals(geogWKT.STAsBinary(), geoWKB));
+    }
+
+    @Test
+    public void testLineString2() throws SQLException {
+        String geoWKT = "LINESTRING EMPTY";
+        byte[] geomCLR = hexStringToByteArray("FFFF00000104000000000000000001000000FFFFFFFFFFFFFFFF02");
+        byte[] geogCLR = hexStringToByteArray("E61000000104000000000000000001000000FFFFFFFFFFFFFFFF02");
+        byte[] geoWKB = hexStringToByteArray("010200000000000000");
+        Geometry geomWKT = Geometry.deserialize(geomCLR);
+        Geography geogWKT = Geography.deserialize(geogCLR);
         assertEquals(geomWKT.asTextZM(), geoWKT);
         assertEquals(geogWKT.asTextZM(), geoWKT);
+        assertTrue(Arrays.equals(geomWKT.STAsBinary(), geoWKB));
+        assertTrue(Arrays.equals(geogWKT.STAsBinary(), geoWKB));
     }
 
     @Test
-    public void testPolygonWkb() throws SQLException {
+    public void testPolygon() throws SQLException {
         String geoWKT = "POLYGON((0 0, 0 3, 3 3, 3 0, 0 0), (1 1, 1 2, 2 1, 1 1))";
-        byte[] geomWKB = hexStringToByteArray(
+        byte[] geomCLR = hexStringToByteArray(
                 "000000000104090000000000000000000000000000000000000000000000000000000000000000000840000000000000084000000000000008400000000000000840000000000000000000000000000000000000000000000000000000000000F03F000000000000F03F000000000000F03F00000000000000400000000000000040000000000000F03F000000000000F03F000000000000F03F020000000200000000000500000001000000FFFFFFFF0000000003");
-        byte[] geogWKB = hexStringToByteArray(
+        byte[] geogCLR = hexStringToByteArray(
                 "E61000000200090000000000000000000000000000000000000000000000000008400000000000000000000000000000084000000000000008400000000000000000000000000000084000000000000000000000000000000000000000000000F03F000000000000F03F0000000000000040000000000000F03F000000000000F03F0000000000000040000000000000F03F000000000000F03F020000000100000000010500000001000000FFFFFFFF0000000003");
-        Geometry geomWKT = Geometry.deserialize(geomWKB);
-        Geography geogWKT = Geography.deserialize(geogWKB);
+        byte[] geoWKB = hexStringToByteArray(
+                "01030000000200000005000000000000000000000000000000000000000000000000000000000000000000084000000000000008400000000000000840000000000000084000000000000000000000000000000000000000000000000004000000000000000000F03F000000000000F03F000000000000F03F00000000000000400000000000000040000000000000F03F000000000000F03F000000000000F03F");
+        Geometry geomWKT = Geometry.deserialize(geomCLR);
+        Geography geogWKT = Geography.deserialize(geogCLR);
+        assertEquals(geomWKT.asTextZM(), geoWKT);
+        assertEquals(geogWKT.asTextZM(), geoWKT);
+        assertTrue(Arrays.equals(geomWKT.STAsBinary(), geoWKB));
+        assertTrue(Arrays.equals(geogWKT.STAsBinary(), geoWKB));
+    }
+
+    @Test
+    public void testPolygon2() throws SQLException {
```

Comment:
testPolygonEmpty()

Reply:
why not take the suggestion above to rename all the testBlah2 tests to more meaningful names?

Reply:
I wanted to stick to numberings because geometrycollection has up to 50 different ones, but I'll change these to "empty" at the end since you insist.